proxy/plugin/cache/store/disk.py

```diff
@@ -22,20 +25,70 @@
 
 
 class OnDiskCacheStore(CacheStore):
+    lock = multiprocessing.Lock()
 
     def __init__(self, uid: UUID, cache_dir: str) -> None:
         super().__init__(uid)
         self.cache_dir = cache_dir
-        self.cache_file_path: Optional[str] = None
+        if not os.path.isdir(self.cache_dir):
+            os.mkdir(self.cache_dir)
+        logger.debug("Opening cache list")
+        self.cache_list = open(os.path.join(self.cache_dir, 'list.txt'), 'at')
```

Comment:
Going forward we must remove hardcoded `list.txt`.  This ideally should be user provided.  Please leave a `TODO` for now or address it by reading the path from an environment variable e.g. `PROXY_PY_CACHE_FILENAME`.  When unavailable, fallback to `list.txt`.

Reply:
Yes, I agree, I will add a `TODO`.

Reply:
jFYI, yesterday I merged https://github.com/abhinavsingh/proxy.py/pull/438 which will soon allow plugins to define custom flags.  Will be migrating configurable parameters within various plugins _(e.g. --blacklist-ip-addresses, --cache-dir, ....)_ as flag this week.  We should be able to re-use this mechanism for `list.txt`.

Talking of `list.txt`, at some point in future far away :) we must also revisit cache schema to ensure best performance under various scenarios _(long running instances, gigabytes of cache data, ....)_